common/commands.h

```diff
@@ -78,37 +76,70 @@ namespace commands::assets {
 
     public:
         using pack::Node::Node;
-        META(In, address, protocol, port, settings);
+        META(CommonIn, address, protocol, port, settings);
     };
+}
 
-    class Return : public pack::Node
+namespace commands::assets {
+    static constexpr const char* Subject = "assets";
+
+    using In = commands::CommonIn;
+
+    using Ext = pack::ObjectList<pack::StringMap>;
+
+    class Asset : public pack::Node
     {
     public:
-        class Asset : public pack::Node
-        {
-        public:
-            //pack::String                      name    = FIELD("name");
-            pack::String                      type    = FIELD("type");
-            pack::String                      subtype = FIELD("sub_type");
-            pack::ObjectList<pack::StringMap> ext     = FIELD("ext");
+        pack::String name    = FIELD("name");   // TBD Needed ???
+        pack::String type    = FIELD("type");
+        pack::String subtype = FIELD("sub_type");
+        Ext          ext     = FIELD("ext");
 
-        public:
-            using pack::Node::Node;
-            META(Asset, type, subtype, ext);
-        };
+    public:
+        using pack::Node::Node;
+        META(Asset, type, subtype, ext);
+    };
 
+    class Return : public pack::Node
+    {
     public:
-        pack::String subAddress = FIELD("sub_address", "-1");
-        Asset        asset      = FIELD("asset");
+        pack::String            subAddress = FIELD("sub_address", "-1");
+        Asset                   asset      = FIELD("asset");
+        pack::ObjectList<Asset> sensors    = FIELD("sensors");
 
     public:
         using pack::Node::Node;
-        META(Return, subAddress, asset);
+        META(Return, subAddress, asset, sensors);
     };
 
     using Out = pack::ObjectList<Return>;
 } // namespace commands::assets
 
 // =====================================================================================================================
 
+// =====================================================================================================================
+
+namespace commands::discoveryauto {
+    static constexpr const char* Subject = "discovery-auto";
+
+    using In = commands::CommonIn;
+
+    class Out : public pack::Node
+    {
+    public:
+        pack::String result = FIELD("result");
+
+    public:
+        using pack::Node::Node;
+        META(Out, result);
+    };
+
+} // namespace commands::discoveryauto
+
+namespace commands {
+    using InVar = pack::Variant<commands::discoveryauto::In, commands::assets::In>;
```

Comment:
Very strange, i'm not sure that it needed here.

Reply:
Take into account